src/test/java/com/truelayer/java/payments/entities/AuthorizationFlowResponseTests.java

```diff
@@ -50,6 +50,26 @@ public void shouldConvertToAuthorizationFailed() {
         assertDoesNotThrow(sut::asAuthorizationFailed);
     }
 
+    @Test
+    @DisplayName("It should get AuthorizationFlow from Authorizing for retro-compatibility")
+    public void shouldGetAuthorizationFlowFromAuthorizing() {
+        AuthorizationFlowResponse sut = new AuthorizationFlowAuthorizing();
```

Comment:
I think here we should use the concrete class on init: 

```suggestion
        AuthorizationFlowAuthorizing sut = new AuthorizationFlowAuthorizing();
```

Reply:
This doesn't seem to be a problem. Are we sure we want to use the concrete class? I think using the abstract class is closer to the common use of this type.